Framework/Geometry/src/Instrument/ComponentInfoBankHelpers.cpp

```diff
@@ -75,6 +75,26 @@ bool isSaveableBank(const ComponentInfo &compInfo, const size_t idx) {
   return false;
 }
 
+/*Function: isAncestorOf. Finds all ancestors up to the root of a component
+ * index and returns true if the possible ancestor is encountered in the
+ * search.
+ *
+ * @param compInfo : Geometry::ComponentInfo Instrument cache containing the
+ * component info.
+ * @param possibleAncestor : the queried ancestor.
+ * @param current : the queried descendant.
+ */
+bool isAncestorOf(const ComponentInfo &compInfo, const size_t possibleAncestor,
+                  const size_t current) {
+  size_t next = current;
+  do {
+    if (compInfo.parent(next) == possibleAncestor)
+      return true;
+    next = compInfo.parent(next);
```

Comment:
I think this might contain a bug. Say we have a chain of ancestors

`1 -> 4 ->6-> 9 -> 15` where `15` is root. 

We input `1` as `current` and `9` as `possibleAncestor`

```
next = 1
4 == 9 //test
next = 4

4 == 9 // test
next = 6

6 == 9
next = 6
```
This is OK. But what happens when I set `possibleAncestor` to `4` and `current` to `4`. Returns false, but should be true.

Also. duplicate calls to `compInfo.parent(next)` seem uncessary.

**Please write for bug and then fix test**

Reply:
refactored

Reply:
test added